src/iterator.js

```diff
@@ -0,0 +1,102 @@
+/**
+  * Builds an iterator that outputs the items from the inputted object
+  * @constructor
+  * @param {*} obj must support iter protocol (has __iter__ and next methods), if sentinel defined:
+  * obj must be callable
+  * @param {*=} sentinel optional if defined returns an object that makes a call to obj until
+  * sentinel is reached
+  * @extends Sk.builtin.object
+  *
+  * @description
+  * Constructor for Python iterator.
+  *
+  */
+Sk.builtin.iterator = function (obj, sentinel) {
+    var objit;
+    if (obj instanceof Sk.builtin.generator) {
+        return obj;
+    }
+    objit = Sk.abstr.lookupSpecial(obj, "__iter__");
+    if (objit) {
+        return Sk.misceval.callsim(objit, obj);
+    }
+    this.sentinel = sentinel;
+    this.flag = false;
+    this.idx = 0;
+    this.obj = obj;
+    if (sentinel === undefined) {
+        this.getitem = Sk.abstr.lookupSpecial(obj, "__getitem__");
```

Comment:
Again, doesn't work with `__getattr__`-supplied `__getitem__`; should probably be fixed at all call sites of `lookupSpecial()`.